### PR TITLE
Task 5

### PR DESCRIPTION
**Pull request by UlybinSergej**

### CODE REVIEW

File: src/task5/task1/Figure.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
+package task5.task1;
+
+public abstract class Figure {
+    int x;
+    int y;
+
+    abstract double square();
```
Comment (YanBelikov) [inline src/task5/task1/Figure.java:7]: As i understand correctly you should to realize abstarct method "square" which should calculates the area of a shape.

File: src/task5/task1/Figure.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+            return Quadrant.II;
+        } else if (x < 0 && y < 0) {
+            return Quadrant.III;
+        } else if (x > 0 && y < 0) {
+            return Quadrant.IV;
+        } else if (x == 0 && y == 0) {
+            return Quadrant.ORIGIN_OF_COORDINATES;
+        } else {
+            return (x == 0) ? Quadrant.THE_COORDINATE_IS_ON_THE_X_AXIS : Quadrant.THE_COORDINATE_IS_ON_THE_Y_AXIS;
+        }
```
Comment (YanBelikov) [inline src/task5/task1/Figure.java:27]: Where is enum?
You should use switch/case here

File: src/task5/task1/Quadrant.java [hunk 3]
```diff
@@ -0,0 +1,5 @@
+package task5.task1;
+
+public enum Quadrant {
+    I, II, III, IV, ORIGIN_OF_COORDINATES, THE_COORDINATE_IS_ON_THE_X_AXIS, THE_COORDINATE_IS_ON_THE_Y_AXIS;
+}
```
Comment (YanBelikov) [inline src/task5/task1/Quadrant.java]: please put it in class where it shoul be used

File: src/task5/task1/Rectangle.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    public Rectangle(int x, int y, int width, int height) throws RectangleSizeException {
+        super(x, y);
+        validateRectangle(width, height);
+        this.width = width;
+        this.height = height;
+    }
+
+    @Override
+    double square() {
+        return width * height;
```
Comment (YanBelikov) [inline src/task5/task1/Rectangle.java:17]: ?

File: src/task5/task1/Rectangle.java [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+    @Override
+    public String toString() {
+        return "Rectangle, square = " + square();
+    }
+
+    private void validateRectangle(int width, int height) throws RectangleSizeException {
+        if (width > 20 || width < 0 || height > 20 || height < 0) {
+            throw new RectangleSizeException("Ширина/Высота вне допустимых значений(0 <= side <= 20)");
+        }
```
Comment (YanBelikov) [inline src/task5/task1/Rectangle.java]: it is correct validation for square, but you should also verify that it is rectangle

File: src/task5/task1/FigureMain.java [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+            Figure f1 = new Circle(4, 8, 9);
+            System.out.println(f1.toString());
+            Figure f2 = new Rectangle(1, 6, 15, 5);
+            System.out.println(f2.toString());
+            Figure f3 = new Circle(1, 1, 7);
+            System.out.println(f3.toString());
+            Figure f4 = new Circle(0, 0, 18);
+            System.out.println(f4.toString());
+            Figure f5 = new Rectangle(15, 45, 50, 84);
+            System.out.println(f5.toString());
```
Comment (YanBelikov) [inline src/task5/task1/FigureMain.java:15]: it is not correct according to task

File: src/task5/task2/Names.java [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+    Артем(7),
+    Николай(8),
+    Константин(9);
+
+    public int nameNumber;
+
+    Names(int nameNumber) {
+        this.nameNumber = nameNumber;
+    }
+}
```
Comment (YanBelikov) [inline src/task5/task2/Names.java]: it should be in class Person

File: src/task5/task1/Figure.java (hunk 2 above)
Comment (UlybinSergej) [reply src/task5/task1/Figure.java:27]: The lecture said that the use enums - best practise

File: src/task5/task1/FigureMain.java (hunk 6 above)
Comment (UlybinSergej) [reply src/task5/task1/FigureMain.java:15]: If you mean fourth and fifth figure - this is for checking validation

File: src/task5/task1/Quadrant.java (hunk 3 above)
Comment (UlybinSergej) [reply src/task5/task1/Quadrant.java]: Done

File: src/task5/task1/Rectangle.java (hunk 4 above)
Comment (UlybinSergej) [reply src/task5/task1/Rectangle.java:17]: height rename into length?
Or error in another?

File: src/task5/task1/Rectangle.java (hunk 5 above)
Comment (UlybinSergej) [reply src/task5/task1/Rectangle.java]: Done

File: src/task5/task2/Names.java (hunk 7 above)
Comment (UlybinSergej) [reply src/task5/task2/Names.java]: Done

**Review by YanBelikov — APPROVED** [review]

**Pull request merged**